### PR TITLE
Refactor to tf12

### PR DESCRIPTION
**Pull request by lukaspour**
I was working on some minor changes to make TF12 work for this module. Could you please review this PR? I think there are plenty of things which could be simplified with TF12. Thanks.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -3,139 +3,169 @@
 # ------------------------------------------------------------------------------
 
 variable "apply_immediately" {
-  type        = "string"
   description = "Specifies whether any database modifications are applied immediately, or during the next maintenance window"
+  type        = string
```
Comment (itsdalmo) [inline variables.tf]: Ideally we'd just use proper types as part of the upgrade. So this one should be `bool` and default to `false` instead of `"false"`?

File: variables.tf [hunk 2]
```diff
@@ -3,139 +3,169 @@
...
   default     = "false"
 }
 
 variable "name_prefix" {
   description = "A prefix used for naming resources."
+  type        = string
+  default     = null
```
Comment (itsdalmo) [inline variables.tf]: This did not use to have a default value, so I don't think it should be added here.

File: variables.tf [hunk 3]
```diff
@@ -3,139 +3,169 @@
...
 }
 
 variable "vpc_id" {
   description = "The VPC ID."
+  type        = string
+  default     = null
```
Comment (itsdalmo) [inline variables.tf]: Same here, this variable is not optional.

File: variables.tf [hunk 4]
```diff
@@ -3,139 +3,169 @@
...
 }
 
 variable "subnet_ids" {
   description = "A list of VPC subnet IDs."
-  type        = "list"
+  type        = list(string)
+  default     = null
```
Comment (itsdalmo) [inline variables.tf]: Not optional 😅 In general, I don't think we should add default values where there previous were not default values, at least not as part of the `0.12` upgrade (we can do it in a separate PR if required)

**Review by itsdalmo — CHANGES_REQUESTED** [review]
Awesome stuff! Only a couple of considerations on my side regarding variables, please see comments 🙌

File: variables.tf (hunk 1 above)
Comment (lukaspour) [reply variables.tf]: I was not sure if the input is string and value is "false" as an arbitrary string, or it is a real bool value, thanks! :-)

File: variables.tf (hunk 2 above)
Comment (lukaspour) [reply variables.tf]: I had to check the documentation for this but imho I thought null will enforce the value to be defined by a user of the module the same way not filling the default value did the same in past versions.

I have tested that by not giving `name_prefix` any value and ended up with plenty of:
```
Error: Invalid template interpolation value

  on terraform-aws-rds-instance/main.tf line 46, in resource "aws_db_subnet_group" "main":
  46:   name        = "${var.name_prefix}-subnet-group"
    |----------------
    | var.name_prefix is null

The expression result is null. Cannot include a null value in a string
template.
```

File: variables.tf (hunk 4 above)
Comment (lukaspour) [reply variables.tf]: Yes, but there is plenty of things we can just leave out and work with them the same way we did before. For example, all outputs/inputs could also be written as "${var.variable}" and tf12 accepts it anyway, also, in many cases, types do not need to be strictly defined as well.

It depends, what would you prefer?

File: variables.tf (hunk 2 above)
Comment (itsdalmo) [reply variables.tf]: Setting `default` means that the variable becomes optional. I'm assuming in the example above, you did not supply `name_prefix` as input to the module, and since you've set the default to be `null`, you get that error message.

If you remove the default value for `name_prefix`, the module will complain about missing a required input instead, which is how it used to work, and what we want :)

**Comment by lukaspour** [comment]
@itsdalmo defaults are fixed now the way you suggested

File: variables.tf [hunk 5]
```diff
@@ -3,139 +3,164 @@
...
   description = "The ARN for the IAM role that permits RDS to send Enhanced Monitoring metrics to CloudWatch Logs"
+  type        = string
   default     = ""
 }
 
 variable "storage_encrypted" {
   description = "Specifies whether the DB instance is encrypted."
-  default     = "true"
+  type        = bool
+  default     = false
```
Comment (itsdalmo) [inline variables.tf]: This should keep the `true` as default value. Can you fix?

**Review by itsdalmo — APPROVED** [review]
LGTM
